dymos/examples/robot_arm/robot_arm_ode.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import openmdao.api as om
+
+
+class RobotArmODE(om.ExplicitComponent):
+
+    def initialize(self):
+        self.options.declare('num_nodes', types=int)
+        self.options.declare('arm_length', types=float, default=5.0)
+
+    def setup(self):
+        nn = self.options['num_nodes']
+        L = self.options['arm_length']
+
+        # inputs: 6 states and 3 controls
+        self.add_input(name='x0', val=np.ones(nn), desc='distance the arm protrudes from the pivot', units='m')
+        self.add_input(name='x1', val=np.ones(nn), desc='horizontal angle of the arm', units='rad')
+        self.add_input(name='x2', val=np.ones(nn), desc='vertical angle of the arm', units='rad')
+        self.add_input(name='x3', val=np.ones(nn), desc='rate of change of arm protrusion', units='m/s')
+        self.add_input(name='x4', val=np.ones(nn), desc='horizontal angular velocity', units='rad/s')
+        self.add_input(name='x5', val=np.ones(nn), desc='vertical angular velocity', units='rad/s')
+
+        self.add_input(name='u0', val=np.ones(nn), desc='arm length control', units='m**2/s**2')
+        self.add_input(name='u1', val=np.ones(nn), desc='horizontal angle control', units='m**3*rad/s**2')
+        self.add_input(name='u2', val=np.ones(nn), desc='vertical angle control', units='m**3*rad/s**2')
+
+        # outputs: 6 state rates
+        self.add_output(name='x0_dot', val=np.ones(nn), desc='rate of change of arm protrusion', units='m/s')
+        self.add_output(name='x1_dot', val=np.ones(nn), desc='horizontal angular velocity', units='rad/s')
+        self.add_output(name='x2_dot', val=np.ones(nn), desc='vertical angular velocity', units='rad/s')
+        self.add_output(name='x3_dot', val=np.ones(nn), desc='second derivative of arm protrusion', units='m/s**2')
+        self.add_output(name='x4_dot', val=np.ones(nn), desc='horizontal angular acceleration', units='rad/s**2')
+        self.add_output(name='x5_dot', val=np.ones(nn), desc='vertical angular acceleration', units='rad/s**2')
+
+        # set up the partials
+        r = c = np.arange(nn)
+
+        self.declare_partials(of='x0_dot', wrt='x0', rows=r, cols=c, val=0.0)
+        self.declare_partials(of='x0_dot', wrt='x1', rows=r, cols=c, val=0.0)
+        self.declare_partials(of='x0_dot', wrt='x2', rows=r, cols=c, val=0.0)
+        self.declare_partials(of='x0_dot', wrt='x3', rows=r, cols=c, val=1.0)
+        self.declare_partials(of='x0_dot', wrt='x4', rows=r, cols=c, val=0.0)
+        self.declare_partials(of='x0_dot', wrt='x5', rows=r, cols=c, val=0.0)
+        self.declare_partials(of='x0_dot', wrt='u0', rows=r, cols=c, val=0.0)
+        self.declare_partials(of='x0_dot', wrt='u1', rows=r, cols=c, val=0.0)
```

Comment:
For derivatives that are 0.0, it is more efficient to just not declare them at all.

Reply:
Fixed that but why is that the case? Is it a coloring issue?

Reply:
When you define it, it creates an entry in the jacobian dictionary that is an r x c arrary of zeros, and it performs a matrix vector product because it sees 0,0 as just another number that needs to be multiplied.  If you don't declare it, then no entry is even created for it, so it does nothing for that input/output pair.

Reply:
Makes sense. Thanks!